### PR TITLE
Feature/sg 609 add ddp readme

### PR DESCRIPTION
**Pull request by Louis-Dupont**
Note sure about the different titles, and how to organize the different sections, open for discussions

### CODE REVIEW

File: documentation/source/Device.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+from super_gradients.training.utils.distributed_training_utils import setup_device
+
+# Launch DP on 4 GPUs'
+setup_device(multi_gpu='DP', num_gpus=4)
+
+# Unchanged
+trainer = Trainer(...)
+trainer.train(...)
+```
+**Tip**: To optimize running time we recommend to call `setup_device` as early as possible.
```
Comment (shaydeci) [inline documentation/source/Device.md:50]: runtime*

File: documentation/source/Device.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+from super_gradients.training.utils.distributed_training_utils import setup_device
+
+# Launch DDP on 4 GPUs'
+setup_device(num_gpus=4) # Equivalent to: setup_device(multi_gpu='DDP', num_gpus=4)
+
+# Unchanged
+trainer = Trainer(...)
+trainer.train(...)
+```
+**Tip**: To optimize running time we recommend to call `setup_device` as early as possible.
```
Comment (shaydeci) [inline documentation/source/Device.md:69]: *runtime

File: documentation/source/Device.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# Training modes
+
+SuperGradients allows users to train models on different modes:
+- CPU
+- single GPU - (CUDA)
+- single GPU - Data Parallel (DP)
+- multiple GPUs' - Distributed Data Parallel (DDP)
+
```
Comment (shaydeci) [inline documentation/source/Device.md:8]: Please give some intro on DP and DDP followed by torch links for more details.

File: documentation/source/Device.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+# Unchanged
+trainer = Trainer(...)
+trainer.train(...)
+```
+**Tip**: To optimize running time we recommend to call `setup_device` as early as possible.
+
+
+
+### DDP - Distributed Data Parallel
```
Comment (shaydeci) [inline documentation/source/Device.md:54]: I think a section about DDP/Devices when training with configs is important as well.

**Review by shaydeci — CHANGES_REQUESTED** [review]
Comments inline.

File: documentation/source/Device.md (hunk 1 above)
Comment (Louis-Dupont) [reply documentation/source/Device.md]: fixed

File: documentation/source/Device.md (hunk 2 above)
Comment (Louis-Dupont) [reply documentation/source/Device.md]: fixed

File: documentation/source/Device.md (hunk 4 above)
Comment (Louis-Dupont) [reply documentation/source/Device.md]: Its pretty short: https://github.com/Deci-AI/super-gradients/blob/7e6af4cc70e0b27c7070f7e3f534a07aaa996dc6/documentation/source/Device.md?plain=1#L216-L229

File: documentation/source/Device.md (hunk 3 above)
Comment (Louis-Dupont) [reply documentation/source/Device.md:8]: DP:
https://github.com/Deci-AI/super-gradients/blob/7e6af4cc70e0b27c7070f7e3f534a07aaa996dc6/documentation/source/Device.md?plain=1#L35-L54

DDP:
https://github.com/Deci-AI/super-gradients/blob/7e6af4cc70e0b27c7070f7e3f534a07aaa996dc6/documentation/source/Device.md?plain=1#L74-L87

**Review by shaydeci — APPROVED** [review]
LGTM
